src/esrs/policy.rs

```diff
@@ -0,0 +1,56 @@
+use async_trait::async_trait;
+
+use crate::{AggregateManager, StoreEvent};
+
+/// This trait is used to implement a `Policy`. A policy is intended to be an entity where to put
+/// non-transactional side effects.
+#[async_trait]
+pub trait Policy<Manager>
+where
+    Self: PolicyClone<Manager>,
```

Comment:
No, you are right. Maybe it is not needed. But i think that this would help developer instead of let him have the duty to put everything under an `Arc`. Moreover i think that until the complexity is hidden in the library there's no problem of give a feature to caller. But maybe is just me. @angelo-rendina-prima and @matteosister wdyt?

Reply:
I think the problem with this feature is once you clone it you can still do `add_projector` and `add_policy`, so we'll have the same statements but a different store which may cause confusion.
It would be easier that once you've run setup you cannot change it, or if you can to be able to do it, to do it from one place.

Reply:
This is a good point. I'll remove that :)